pkg/catalog/cleanup.go

```diff
@@ -47,27 +46,24 @@ func DeleteIncorrectPluginEntriesFromCatalog() {
 }
 
 // MigrateContextPluginsAsStandaloneIfNeeded updates the catalog cache to move all the
-// context-scoped plugins associated with the active context as standalone plugins
+// context-scoped plugins as standalone plugins
 // and removes the context-scoped plugin mapping from the catalog cache.
-// This is to ensure backwards compatibility when user migrates from pre v1.3 version of
-// the CLI, the context-scoped plugins are still gets shown as installed
+// This is to ensure backwards compatibility when the user migrates from pre v1.3 version of
+// the CLI, the context-scoped plugins are still shown as installed
 func MigrateContextPluginsAsStandaloneIfNeeded() {
-	activeContexts, err := configlib.GetAllActiveContextsList()
-	if err != nil || len(activeContexts) == 0 {
-		return
-	}
-
 	c, lockedFile, err := getCatalogCache(true)
 	if err != nil {
 		return
 	}
 	defer lockedFile.Close()
 
-	for _, ac := range activeContexts {
-		for pluginKey, installPath := range c.ServerPlugins[ac] {
+	for _, association := range c.ServerPlugins {
+		for pluginKey, installPath := range association {
 			c.StandAlonePlugins.Add(pluginKey, installPath)
 		}
-		delete(c.ServerPlugins, ac)
 	}
+	// Delete all entries by reassigning to a new empty map
+	c.ServerPlugins = make(map[string]PluginAssociation)
+
 	_ = saveCatalogCache(c, lockedFile)
```

Comment:
What I understand is with this change now we are converting all the ServerPlugins as StandalonePlugins instead of just doing that for the **ActiveContexts**.
Is this change intentional?

Reply:
Correct.  See point 3 in the PR description:

> with this PR, since we only run the plugin migration once, we must migrate the context-scoped plugins for all contexts immediately. This is a slight change in behaviour because context-scoped plugins for an inactive (and possibly old context) will now appear as standalone plugins right away for the new CLI. Since this new behaviour is similar to point 2 above, I feel it is acceptable

Originally, I started by keeping the original solution of checking this migration at every command, but not blindly updating the catalog if there was nothing to migrate.  However this still has a cost compared to running the migration only once, since we don't have to check the catalog for every command, just in case.

But I wanted your opinion if there may be side-effects I missed.

Reply:
> instead of just doing that for the ActiveContexts

To be very clear, just in case, with the previous approach, when the active context would change, we would migrate the plugins for the new active context.  So, assuming every context eventually became active, then all ServerPlugins would eventually be migrated as StandalonePlugins.

With this PR we do it all at once and be done with it.

Do your recall if "only doing it for the active contexts" was a kind of optimization?  Maybe because it was run for every command?